Lib/ufo2ft/outlineCompiler.py

```diff
@@ -873,6 +881,44 @@ def setupTable_post(self):
         post.minMemType1 = 0
         post.maxMemType1 = 0
 
+    def setupTable_COLR(self):
+        """
+        Compile the COLR table.
+
+        **This should not be called externally.**
+        """
+        if "COLR" not in self.tables:
+            return
+
+        from fontTools.ttLib.tables.C_O_L_R_ import LayerRecord
+        layerInfo = self.ufo.lib[COLOR_LAYERS_KEY]
+        colorLayerLists = {}
+        for baseGlyphName, layers in layerInfo.items():
+            colorLayerLists[baseGlyphName] = [
+                    LayerRecord(layerGlyphName, colorID)
+                        for layerGlyphName, colorID in layers]
+        self.otf["COLR"] = colr = newTable("COLR")
+        colr.version = 0
+        colr.ColorLayers = colorLayerLists
+
+    def setupTable_CPAL(self):
+        """
+        Compile the CPAL table.
+
+        **This should not be called externally.**
+        """
+        if "CPAL" not in self.tables:
+            return
+
+        from fontTools.ttLib.tables.C_P_A_L_ import Color
+        palettes = self.ufo.lib[COLOR_PALETTES_KEY]
+        assert len({len(p) for p in palettes}) == 1, "color palettes have different lengths"
```

Comment:
maybe better to raise `ufo2ft.errors.InvalidFontData` here instead of assert, as this lib key could be user defined and as such contain palettes of different lengths by mistake.

Reply:
Done.